File: src/main/java/com/group/libraryapp/domain/url/UrlRepository.java
package com.group.libraryapp.domain.url;

import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface UrlRepository extends JpaRepository<Url, Long> {
    Optional<Url> findByShortKey(String shortKey);
    boolean existsByShortKey(String shortKey);
}

File: src/main/kotlin/com/griotold/mcp_demo/repository/UrlRepository.kt
package com.griotold.mcp_demo.repository

import com.griotold.mcp_demo.domain.Url
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.Optional

@Repository
interface UrlRepository : JpaRepository<Url, Long> {
    fun findByShortKey(shortKey: String): Optional<Url>
    fun existsByShortKey(shortKey: String): Boolean
}
